Gems/PhysX/Code/Tests/CharacterGameplayControllerTests.cpp

```diff
@@ -114,41 +117,136 @@ namespace PhysX
+
+        // Get the end velocity and validate that it is the expected value.
+        auto endVelocity = basis.m_gameplayController->GetFallingVelocity();
+        EXPECT_THAT(endVelocity.GetZ(), testing::FloatNear(expectedVelocity.GetZ(), 0.001f));
+    }
+
+    //! Test the verifies that the Falling Velocity sets and uses the set value to interact with the tick incrementing.
+    TEST_F(PhysXDefaultWorldTest, CharacterGameplayController_FallingVelocitySetsWhileMoving)
+    {
+        const AZ::Vector3 expectedVelocity(0.0f, 0.0f, 22.0f);
+        GameplayTestBasis basis(m_testSceneHandle, DefaultGravityMultiplier, DefaultGroundDetectionBoxHeight);
+
+        // Get the original velocity so we can determine that set actually changes things.
+        auto originalVelocity = basis.m_gameplayController->GetFallingVelocity();
+
+        // Run the scene for a while and validate that the set velocity is working
+        for (int i = 0; i < 10; i++)
+        {
+            // Calculate and set velocity based upon the original velocity, expected velocity, and loop count modifier
+            auto setVelocity = originalVelocity + expectedVelocity + AZ::Vector3(0.0f, 0.0f, float(i));
+            basis.m_gameplayController->SetFallingVelocity(setVelocity);
+
+            basis.Update();
+
+            // Get the endVelocity & current gravity to perform calculations on.
+            auto endVelocity = basis.m_gameplayController->GetFallingVelocity();
+            const auto gravity = basis.m_testScene->GetGravity();
+
+            // The end velocity should be decelerated by gt when g is gravity acceleration and t is the length of time
+            const float expectedDeceleration = gravity.GetZ() * basis.m_timeStep;
+
+            // Verify that the end velocity is the set velocity minus the deceleration for a single time step
+            EXPECT_THAT(endVelocity.GetZ(), testing::FloatNear(setVelocity.GetZ() + expectedDeceleration, 0.001f));
+        }
+    }
+
+    //! Test that verifies that the Ground Detec Set & Get methods function as expected.
+    TEST_F(PhysXDefaultWorldTest, CharacterGameplayController_SetGroundDetectionHeight)
+    {
+        GameplayTestBasis basis(m_testSceneHandle, DefaultGravityMultiplier, DefaultGroundDetectionBoxHeight);
+
+        const float expectedHeight = 10.0f;
+        const auto originalHeight = basis.m_gameplayController->GetGroundDetectionBoxHeight();
+
+        // Set the ground detection to the excpected heigh + the original height to ensure that it changes.
+        basis.m_gameplayController->SetGroundDetectionBoxHeight(expectedHeight + originalHeight);
+
+        // Get the end height and verify that it is set to the expected value set.
+        auto endHeight = basis.m_gameplayController->GetGroundDetectionBoxHeight();
+        EXPECT_THAT(endHeight, testing::FloatNear(expectedHeight + originalHeight, 0.001f));
+    }
+
+    //! Test to determine if the ground will detect while moving.
```

Comment:
nit: (rephrase slightly) _Test to determine if the ground is detected while moving_.

Reply:
Fixed